go-controller/pkg/ovn/pods.go

```diff
@@ -309,8 +331,9 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 					klog.Errorf("Failed to block off already allocated IPs: %s for pod %s on node: %s"+
 						" error: %v", util.JoinIPNetIPs(podIfAddrs, " "), portName,
 						logicalSwitch, err)
-					return fmt.Errorf("failed to block off already allocated IPs %s: for node: %s, error: %v",
-						util.JoinIPNetIPs(podIfAddrs, " "), logicalSwitch, err)
+				} else {
+					// this should also be treated as an allocation for purposes of error handling
+					releaseIPs = true
```

Comment:
I think you need to set `releaseIPs` outside of the `else` arm, right? If it was already allocated and we couldn't block it off, then we should still take the opportunity to release it if we fail again?

Reply:
So long as we are clearing the port addresses, we should technically never hit the condition anymore. But I am ok with moving this outside the `else {}`

Reply:
For the sake of clarity, let us please set it outside the `else {}`. Thanks.

Reply:
done.